### PR TITLE
Bug 2001442: empty termination.log file for the kube-apiserver has too permissive mode

### PR DESCRIPTION
**Pull request by p0lyn0mial**
fix watch-termination to create termination.log file with restricted permissions.

watch-termination uses lumberjack for logging.
it creates permissive files by default 0644 and at the moment there is no way to specify
permission while creating a file, the only way to workaround is to create a file before.

this pr touches a file with restrictive permissions 0600 and relies on the fact that
lumberjack respects and copies permission over if the file already exists.

### CODE REVIEW

File: cmd/watch-termination/main.go [hunk 1]
```diff
@@ -253,6 +253,20 @@ func (w *terminationFileWriter) WriteToTerminationLog(bs []byte) (int, error) {
 	}
 
 	if w.logger == nil {
+		if exist, err := fileExists(w.fn); err != nil {
+			return 0, err
```
Comment (sttts) [inline cmd/watch-termination/main.go:257]: chmod?

Comment (p0lyn0mial) [reply cmd/watch-termination/main.go:257]: it really only applies to new files.
already existing files are changed by the `Fixing audit permissions` container.

Comment (p0lyn0mial) [reply cmd/watch-termination/main.go:257]: which gets executed on every restart

Comment (sttts) [reply cmd/watch-termination/main.go:257]: ack

**Comment by p0lyn0mial** [comment]
/hold

we will merge it after rebase lands

**Comment by p0lyn0mial** [comment]
> the rebase PR merged - https://github.com/openshift/kubernetes/pull/1087

/hold cancel

**Pull request merged**